packages/dart/npt_flutter/lib/features/onboarding/util/atsign_manager.dart

```diff
@@ -19,12 +44,119 @@ abstract class AtsignInformation {
 // Now you have the rootDomain for the existing atSign and can use it to onboard
 // correctly
 
-Future<Map<String, AtsignInformation>> getAtsignEntries() {
-  return Future.value({});
+Future<Map<String, AtsignInformation>> getAtsignEntries() async {
+  var keychainAtSigns = await KeychainUtil.getAtsignList() ?? [];
+  var atSignInfo = <AtsignInformation>[];
+  try {
+    atSignInfo = await _getAtsignInformationFromFile();
+  } catch (e) {
+    App.log(
+      "Failed get Atsign Information, ignoring invalid file: ${e.toString()}"
+          .loggable,
+    );
+    return {};
+  }
+  var atSignMap = <String, AtsignInformation>{};
+  for (var item in atSignInfo) {
+    if (keychainAtSigns.contains(item.atSign)) {
+      atSignMap[item.atSign] = item;
+    }
+  }
+  return atSignMap;
 }
 
 // This class will allow you to store atSign information
 // you need to call this after onboarding a NEW atSign
-Future<bool> saveAtsignInformation(AtsignInformation info) {
-  return Future.value(true);
+Future<bool> saveAtsignInformation(AtsignInformation info) async {
```

Comment:
When we onboard, make sure to save the AtsignInformation (atSign and rootDomain) so it can be retrieved later